libs/core/langchain_core/runnables/base.py

```diff
@@ -567,6 +613,40 @@ async def ainvoke(
         coros = map(ainvoke, inputs, configs)
         return await gather_with_concurrency(configs[0].get("max_concurrency"), *coros)
 
+    async def abatch_as_completed(
+        self,
+        inputs: List[Input],
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        *,
+        return_exceptions: bool = False,
+        **kwargs: Optional[Any],
+    ) -> AsyncIterator[Tuple[Input, Output]]:
```

Comment:
Do we need this to be `AsyncIterator[Tuple[int, Input, Output]]`?

Duplicated Input, Output but different configs

Reply:
like index in input array?

Reply:
in that case I'd actually prefer to not return the input at all, just index and output, wdyt?

Reply:
yeah that works